### ISSUE TITLE
Suppress nullability warnings after Is.Not.Null constraint

### ISSUE THREAD

**Issue by Dreamescaper** [nunit/nunit.analyzers#157]
As discussed here https://github.com/nunit/nunit/issues/3376#issuecomment-535879642, it would be great to suppress nullability warnings after Is.Not.Null asserts, e.g.
```csharp
string? str = GetNullable();
Assert.That(str, Is.Not.Null);
Assert.That(str.Length, Is.Not.Zero);  // <----- should be no warning here
```

Unfortunately, we cannot impact roslyn's nullability analysis, only suppress analyzers (see https://github.com/dotnet/roslyn/issues/32042#issuecomment-540524650), which means that at least very basic logic should be re-invented.

Cases that I think should be covered:
`Assert.That(obj?.PropA?.PropB, Is.Not.Null)`
`Assert.That(obj, Is.Not.Null & Is.Not.Empty)`
`Assert.That(obj, Is.Not.Null.And.Not.Empty)`

```csharp
Assert.Multiple(() =>
{
	string? str = GetNullable();
	Assert.That(str, Is.Not.Null);
	Assert.That(str.Length, Is.Not.Zero);  // <----- warning should remain here
});
```

**Comment by mikkelbu** [comment]
I can definitely see the benefit of this, but also the complexity if we want to handle more than the most basic cases, as we would have to implement the same flow analysis as in the compiler.

So the easiest is probably to start with the most basic example and then build upon this.

**Comment by Dreamescaper** [comment]
The only concern is - doing that would probably require to drop VS2017 support.
I don't know any way to reference Roslyn 3.0 package _and_ maintain compatibility with VS2017.

**Comment by manfred-brands** [comment]
I'm looking into the use of a DiagnosticsSuppressor.
